src/Ansatz/Chain.jl

```diff
@@ -223,42 +222,43 @@ function Base.rand(rng::AbstractRNG, ::Type{A}, ::Type{B}, ::Type{S}; kwargs...)
     return rand(rng, ChainSampler{B,S}(; kwargs...), B, S)
 end
 
+# TODO let choose the orthogonality center
 function Base.rand(rng::Random.AbstractRNG, sampler::ChainSampler, ::Type{Open}, ::Type{State})
     n = sampler.parameters.n
     χ = sampler.parameters.χ
     p = get(sampler.parameters, :p, 2)
     T = get(sampler.parameters, :eltype, Float64)
 
-    arrays::Vector{AbstractArray{T,N} where {N}} = map(1:n) do i
-        χl, χr = let after_mid = i > n ÷ 2, i = (n + 1 - abs(2i - n - 1)) ÷ 2
-            χl = min(χ, p^(i - 1))
-            χr = min(χ, p^i)
+    arrays = Vector{AbstractArray{T,N} where {N}}()
 
-            # swap bond dims after mid and handle midpoint for odd-length MPS
-            (isodd(n) && i == n ÷ 2 + 1) ? (χl, χl) : (after_mid ? (χr, χl) : (χl, χr))
-        end
+    # Left boundary tensor
+    F = lq!(rand(rng, T, 1, p * min(χ, p)))
```

Comment:
Why `lq!` and not `qr!`?

Reply:
`lq!` is useful here since we want the `Chain` to be in the right-canonical form. We could do it with `qr!` too but it would involve some permutations and reshapes, which is much elegantly solved with `lq!`.

Reply:
If we changed `lq!` for `qr!` and we reshaped the physical leg with the left bond, we would get a left-canonical `Chain`.